inst/include/RcppArmadilloAs.h

```diff
@@ -87,23 +89,70 @@ namespace traits {
             const int  RTYPE = Rcpp::traits::r_sexptype_traits<T>::rtype;
         
             IntegerVector dims = mat.slot("Dim");
-            IntegerVector i = mat.slot("i") ;
-            IntegerVector p = mat.slot("p") ;     
-            Vector<RTYPE> x = mat.slot("x") ;
-            
-            // Creating an empty SpMat            
-            arma::SpMat<T> res((unsigned) dims[0], (unsigned) dims[1]);
+            int nrow = dims[0];
+            int ncol = dims[1];
             
-            // Making space for the elements
-            res.mem_resize((unsigned) x.size());
+            // Creating an empty SpMat
+            arma::SpMat<T> res(static_cast<unsigned>(nrow), static_cast<unsigned>(ncol));
             
-            // Copying elements
-            std::copy(i.begin(), i.end(), arma::access::rwp(res.row_indices));
-            std::copy(p.begin(), p.end(), arma::access::rwp(res.col_ptrs));
-            std::copy(x.begin(), x.end(), arma::access::rwp(res.values));
+            // Get the type of sparse matrix
+            std::string type = Rcpp::as<std::string>(mat.slot("class"));
+            if (type == "dgCMatrix"){
```

Comment:
Stylistically... @eddelbuettel will complain that you did not put a space between `)` and `{`.